### PR TITLE
chore: update database migration ADR

### PR DESCRIPTION
**Pull request by sjaanus**
## Key Updates in the ADR

- **Separation of Migrations in PRs**: Migrations are now required to be carried out in separate pull requests. This change is intended to improve the monitoring and management of database schema changes during deployment.

- **Primary Key Requirement for New Tables**: A new paragraph mandates the inclusion of primary keys in all new tables, emphasizing the importance of data integrity, efficient data retrieval, and supporting table relationships. Additionally, by adding primary keys, we resolve the issue of migrations failing during upgrades in replicated database setups, as we are not using PostgreSQL replica identities. Exceptions to this rule require a compelling justification.

Also added better structuring and styling to ADR for better readability.

### CODE REVIEW

**Review by kwasniew — APPROVED** [review]
Can you elaborate on the "Additionally, by adding primary keys, we resolve the issue of migrations failing during upgrades in replicated database setups, as we are not using PostgreSQL replica identities" in the actual ADR?

File: website/docs/contributing/ADRs/back-end/breaking-db-changes.md [hunk 1]
```diff
@@ -11,19 +11,32 @@ The very same problem occurs when you apply a breaking migration just before the
...
+- **Remove the old DB schema** once no clients use the old version.
 
-Action for a code reviewer:
-* when you spot a migration with `ALTER table DROP COLUMN` or `ALTER table RENAME TO` please raise a flag if the "expand phase" was missed
+### Code Reviewer Responsibilities
+- **Action for a Code Reviewer:** When you spot a migration with `ALTER TABLE DROP COLUMN` or `ALTER TABLE RENAME TO`, please raise a flag if the "expand phase" was missed.
+
+
+### Separate Migrations as Distinct PRs
+- Carry out all migrations in **separate pull requests (PRs)** and monitor them closely during deployment.
```
Comment (thomasheartman) [inline website/docs/contributing/ADRs/back-end/breaking-db-changes.md:36]: Can we add a note on how we monitor them? I don't know exactly. Would that be keeping an eye on metrics in grafana during rollouts, for instance?

File: website/docs/contributing/ADRs/back-end/breaking-db-changes.md [hunk 2]
```diff
@@ -11,19 +11,32 @@ The very same problem occurs when you apply a breaking migration just before the
...
+
+### Primary Key Requirement for New Tables
+- All new tables should have a primary key to guarantee data integrity, facilitate efficient querying, and enable the establishment of foreign key relationships with other tables.
+
+
```
Comment (thomasheartman) [inline website/docs/contributing/ADRs/back-end/breaking-db-changes.md:41]: Absolute nit: Why is there a double space here? Can we remove one of them? Or if it's meant to separate the content, maybe add a separator line, i.e. "---"?

```suggestion

```

File: website/docs/contributing/ADRs/back-end/breaking-db-changes.md [hunk 3]
```diff
@@ -11,19 +11,32 @@ The very same problem occurs when you apply a breaking migration just before the
...
-* when you spot a migration with `ALTER table DROP COLUMN` or `ALTER table RENAME TO` please raise a flag if the "expand phase" was missed
+### Code Reviewer Responsibilities
+- **Action for a Code Reviewer:** When you spot a migration with `ALTER TABLE DROP COLUMN` or `ALTER TABLE RENAME TO`, please raise a flag if the "expand phase" was missed.
+
+
+### Separate Migrations as Distinct PRs
+- Carry out all migrations in **separate pull requests (PRs)** and monitor them closely during deployment.
+
+### Primary Key Requirement for New Tables
+- All new tables should have a primary key to guarantee data integrity, facilitate efficient querying, and enable the establishment of foreign key relationships with other tables.
```
Comment (thomasheartman) [inline website/docs/contributing/ADRs/back-end/breaking-db-changes.md:39]: Do we have any tables without PKs? Or have we had any?

Comment (sjaanus) [reply website/docs/contributing/ADRs/back-end/breaking-db-changes.md:39]: We had, but not anymore.

**Pull request merged**